test/common/network/listen_socket_impl_test.cc

```diff
@@ -67,8 +67,50 @@ TEST_P(ListenSocketImplTest, BindSpecificPort) {
   EXPECT_EQ(addr->asString(), socket3.localAddress()->asString());
 }
 
+TEST_P(ListenSocketImplTest, BindSpecificPortUdp) {
```

Comment:
Mild preference for making SocketType a new argument to the TEST_P, and having a createTcpListenSocketPtr which switches on type so that rather than copy/paste/editing each test as we add them, we get them for free.

Reply:
@alyssawilk agree. I wanted to do it like I did for the proxy test. Added now. Not sure if this is what you meant.